scripts/wod.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+//pool, explode-at, rote
+//where rote means the intial failed dice are rerolled
+//notably, those rerolled dice cannot explode, and failed exploding dice from qualities can't either
+//meaning that rote has to be counted separately AFTER accounting for dice explosion
+//All 10s explode
+
+module.exports = function(robot) {
+	robot.respond(/roll (\d+)e(\d+)([r])/i, function (msg) {
+		let answer;
+		let pool = parseInt(msg.match[1]);
+    	let explode = parseInt(msg.match[2]);
+    	let rote = parseInt(msg.match[3]);
+    	if (explode == NaN) {
+    		//TODO why NaN?
+    		//TODO test regex
+    		//TODO fill out rest of msg.reply
+    		//TODO bad input cases
+    	}
+	}
+}
+
+
+function roll() {
+	let result = Math.floor(Math.random()*10) + 1;
+	return(result);
+}
+
+
+function rollMany(dice){
+	let results = [];
+	for (var i = 0; i < dice; i++) {
+		results[i] = roll();
```

Comment:
I'd prefer `results.push(roll())` to implicit-array-extending-through-assignment.